wp-content/plugins/pri-migration-helper/cli/includes/class-pmh-worker.php

```diff
@@ -886,5 +886,169 @@ public function post_tags_fix_duplicate_page( $args ) {
 		// Return false if no post_tags found.
 		return count( $a_post_tags ) > 0 ? $a_post_tags[0] : null;
 	}
+
+	/**
+	 * Import extra image fields from Drupal to WordPress.
+	 *
+	 * Example: wp tw-fix image_captions
+	 *
+	 * @param array $a_args
+	 */
+	public function image_captions( $args ) {
+
+		// Simulate running importer.
+		global $fgd2wpp;
+		ob_start();
+		$fgd2wpp->importer();
+		ob_get_clean();
+
+		// Connect.
+		WP_CLI::log( 'Connecting to Drupal.' );
+
+		$connected = $fgd2wpp->drupal_connect();
+
+		// If Drupal connection is established.
+		if ( $connected ) {
+
+			// Show success.
+			WP_CLI::success( "Drupal connection is established." );
+
+		} else {
+
+			// Return error.
+			WP_CLI::error( "Drupal connection is not established." );
+		}
+
+		// Set the minimum post ID
+		$i_start_from = isset( $args[0] ) ? (int) $args[0] : 0;
+		$limit        = isset( $args[1] ) ? (int) $args[1] : 100;
+
+		// Set the last ID.
+		$this->i_last_id = $i_start_from;
+
+		// Get all media ids.
+		$a_media_ids = $this->get_images_by_asc_id( $limit );
+
+		// Start.
+		WP_CLI::log( 'Processing images..' );
+
+		// Loop through all media ids.
+		do {
+
+			// Start.
+			WP_CLI::log( wp_sprintf( 'Running fix for %s images starting from ID: %s', $limit, $this->i_last_id ) );
+
+			if ( ! $a_media_ids ) {
+				break;
+			}
+
+			// Loop through all media ids.
+			foreach ( $a_media_ids as $i_media_id ) {
+
+				$this->_image_captions( array( $i_media_id, $connected ) );
+
+				$this->i_last_id = (int) $i_media_id;
+			}
+
+			// Get all media ids.
+			$a_media_ids = $this->get_images_by_asc_id( 100 );
+
+		} while ( $a_media_ids );
+	}
+
+	/**
+	 * Process images in batches.
+	 *
+	 * @param int $per_process_limit The number of images to process in each batch.
+	 *
+	 * @return array The post ids.
+	 */
+	public function get_images_by_asc_id( $per_process_limit ) {
+
+		// Query all the images.
+		$args = array(
+			'post_type'      => 'attachment',
+			'post_mime_type' => 'image',
+			'post_status'    => 'inherit',
+			'post_parent'    => null,
```

Comment:
Don't we want to include images that have been attached to a post in this fix?

Reply:
Done!